utils/python/CIME/case.py

```diff
@@ -128,13 +128,10 @@ def initialize_derived_attributes(self):
         self.tasks_per_numa = int(math.ceil(self.tasks_per_node / 2.0))
         smt_factor = max(1,int(self.get_value("MAX_TASKS_PER_NODE")/self.get_value("PES_PER_NODE")))
 
-        self.cores_per_task = ((self.get_value("MAX_TASKS_PER_NODE")/smt_factor) \
-                               / self.tasks_per_node) * 2
+        self.cores_per_task = self.thread_count / smt_factor
```

Comment:
@jedwards4b please take a close look at this. I was getting very odd results on melvin with the previous implementation of self.cores_per_task. The prior impl seemed to assume that it had to use every core for a problem. For example, for a 16x2 case on melvin (64 MAX_TASK, 64 PPN), the resulting cores_per_task was 8. This is not correct, at least not for what I intended, because the results TOTAL_CORES came out to 128.